pkg/controller/pod_driver.go

```diff
@@ -104,6 +121,44 @@ func (p *PodDriver) getPodStatus(pod *corev1.Pod) podStatus {
 	return podPending
 }
 
+func (p *PodDriver) checkAnnotations(pod *corev1.Pod) error {
+	// check refs in mount pod, the corresponding pod exists or not
+	lock := config.GetPodLock(pod.Name)
+	lock.Lock()
+	defer lock.Unlock()
+
+	annotation := make(map[string]string)
+	var existTargets int
+	for k, target := range pod.Annotations {
+		if strings.HasPrefix(k, "juicefs-") {
+			deleted, exists := p.mit.deletedPods[getPodUid(target)]
+			if deleted || !exists {
+				// target pod is deleted
+				continue
+			}
+			existTargets++
+		}
+		annotation[k] = target
+	}
+
+	if len(pod.Annotations) != len(annotation) {
+		pod.Annotations = annotation
+		if err := p.Client.UpdatePod(pod); err != nil {
```

Comment:
UpdatePod may fail because resourceVersion of pod in kubelet may be behind to the apiServive.